pyfr/plugins/tavg.py

```diff
@@ -132,20 +162,85 @@ def _eval_acc_exprs(self, intg):
             exprs.append([npeval(v, subs) for v in self.aexprs])
 
         # Stack up the expressions for each element type and return
-        return [np.dstack(exs).swapaxes(1, 2) for exs in exprs]
+        return [np.stack(exs) for exs in exprs]
+
+    def _eval_fun_exprs(self, avars):
 
-    def _eval_fun_exprs(self, intg, accex):
+        # Prepare the substitution dictionary
+        subs = dict(zip(self.anames, avars))
+
+        # Evaluate the function 
+        fexp = [npeval(v, subs) for v in self.fexprs]
```

Comment:
Does `np.array([npeval(v, subs) for v in self.fexprs])` yield the same as the list + stack call?